Fix part of #8015 and Fix #9914: Refactor backend-api services to return domain objects.

nishantwrp opened the pull request:
## Overview
<!--
READ ME FIRST:
Please answer *both* questions below and check off every point from the Essential Checklist!
If there is no corresponding issue number, fill in N/A where it says [fill_in_number_here] below in 1.
-->

1. This PR fixes or fixes part of #8015.
Fixes #9914 
2. This PR does the following: Refactor backend-api services to return domain objects.

## Essential Checklist

- [x] The PR title starts with "Fix #bugnum: ", followed by a short, clear summary of the changes. (If this PR fixes part of an issue, prefix the title with "Fix part of #bugnum: ...".)
- [x] The linter/Karma presubmit checks have passed locally on your machine.
- [x] "Allow edits from maintainers" is checked. (See [here](https://help.github.com/en/github/collaborating-with-issues-and-pull-requests/allowing-changes-to-a-pull-request-branch-created-from-a-fork) for instructions on how to enable it.)
  - This lets reviewers restart your CircleCI tests for you.
- [x] The PR is made from a branch that's **not** called "develop".

## PR Pointers

- Oppiabot will notify you when you don't add a PR_CHANGELOG label. If you are unable to do so, please @-mention a code owner (who will be in the Reviewers list), or ask on [Gitter](https://gitter.im/oppia/oppia-chat).
- For what code owners will expect, see the [Code Owner's wiki page](https://github.com/oppia/oppia/wiki/Oppia%27s-code-owners-and-checks-to-be-carried-out-by-developers).
- Make sure your PR follows conventions in the [style guide](https://github.com/oppia/oppia/wiki/Coding-style-guide), otherwise this will lead to review delays
- Never force push. If you do, your PR will be closed.

nishantwrp committed:
Topics and Skills Dashboard (All checks passing)

nishantwrp committed:
Backend tests

nishantwrp committed:
All other files done (Tests passing)

nishantwrp committed:
Merge remote-tracking branch 'upstream/develop' into refactor-backend-api-services-3

nishantwrp committed:
Fix issues with merge

nishantwrp commented on core/controllers/topics_and_skills_dashboard.py:
```diff
@@ -126,16 +126,12 @@ def get(self):
             if (skill_id not in skill_ids_assigned_to_some_topic) and (
                     skill_id not in merged_skill_ids):
                 untriaged_skill_summary_dicts.append(skill_summary_dict)
-                categorized_skills_dict['untriaged_skills'].append({
-                    'skill_id': skill_id,
-                    'skill_description': skill_summary_dict['description']
-                })
```
I removed this because of two reasons
- `categorized_skills_dict` is a dict with topic names as keys. In typescript if there's a dict all the values should be of same type.
- The same information was already being returned in `untriaged_skill_summary_dicts`.

seanlip replied:
Thanks @nishantwrp, adding @Showtim3 for review since he can advise on whether this change makes sense.

nishantwrp replied:
Ok @seanlip, Please note that to declare a type for this in the backend-api service. This is **necessary**.

seanlip replied:
Yup sorry -- I don't disagree. I think @Showtim3 and you would need to collaborate to make sure that the changes to the data structure look good and are propagated correctly. Thanks!

Showtim3 replied:
Yeah the changes look okay. Tested them locally also.

nishantwrp commented:
### What did I test?

- **subtopic-viewer-backend-api.service.ts** - Didn't change the response of the service. Just the types.
- **topic-viewer-backend-api.service.ts** - Didn't change the response of the service. Just the types.
- **topic-creation-backend-api.service.ts** - Didn't change the response of the service. Just the types.
- **topics-and-skills-dashboard-backend-api.service.ts** - Loaded the dummy data. The topic and skill dashboard page looked fine. The skill merge modal was fine!, the filters were working. Assign to topic was working. Remove skills from topic, subtopic! Works fine!
- **contribution-opportunities-backend-api.service.ts** - Didn't change the response much. But did test the community dashboard. No console errors!
- **exploration-features-backend-api.service.ts** - Tested the places where this service was used.
- **state-top-answers-stats-backend-api.service.ts** - Statistics tab data was displayed correctly.
- **profile-page-backend-api.service.ts** - Tested the profile page. Seems fine! + e2e tests are passing for that page.

nishantwrp committed:
Minor Bug

vojtechjelinek commented on core/templates/components/question-directives/questions-list/questions-list.directive.ts:
```diff
@@ -51,7 +51,7 @@ require('domain/question/QuestionObjectFactory.ts');
 require('domain/skill/skill-backend-api.service.ts');
 require('domain/skill/SkillDifficultyObjectFactory.ts');
-require('domain/skill/SkillSummaryObjectFactory.ts');
+require('domain/skill/ShortSkillSummaryObjectFactory.ts');
```
This should be moved before `require('domain/skill/skill-backend-api.service.ts');`. K is after h in the alphabet ditto below.

nishantwrp replied:
done

vojtechjelinek commented on core/templates/components/skill-selector/skill-selector.directive.html:
```diff
@@ -6,27 +6,27 @@
           </md-card>
           <div ng-if="!checkIfEmpty(skills)" ng-repeat="(subtopicName, skills) in subtopics" class="list-view-item ">
             <md-card class="list-view-item">
               <label style="padding-top:20px;padding-bottom:20px;padding-left:10px"><[subtopicName]></label>
-              <md-radio-button style="margin-left:10px" class="protractor-test-skills-list-item" ng-repeat="skill in skills | filter:skillFilterText" ng-value="skill.skill_id" aria-label="<[skill.skill_description]>">
-                <[skill.skill_description]>
+              <md-radio-button style="margin-left:10px" class="protractor-test-skills-list-item" ng-repeat="skill in skills | filter:skillFilterText" ng-value="skill.id" aria-label="<[skill.description]>">
+                <[skill.description]>
               </md-radio-button>
             </md-card>
           </div>
         </md-card>
-        <md-card ng-if="hasUntriagedSkills(topicName)" ng-repeat="(topicName, skills) in categorizedSkills" class="list-view-item">
+        <md-card ng-if="untriagedSkillSummaries" class="list-view-item">
           <md-card layout="row" class="list-view-item">
             <h2><label style="padding-left:10px">Untriaged Skills</label></h2>
           </md-card>
-          <div ng-if="!checkIfEmpty(skills)" class="list-view-item ">
+          <div ng-if="!checkIfEmpty(untriagedSkillSummaries)" class="list-view-item ">
             <md-card class="list-view-item">
-              <md-radio-button style="margin-left:10px" class="protractor-test-skills-list-item" ng-repeat="skill in skills | filter:skillFilterText" ng-value="skill.skill_id" aria-label="<[skill.skill_description]>">
-                <[skill.skill_description]>
+              <md-radio-button style="margin-left:10px" class="protractor-test-skills-list-item" ng-repeat="skill in untriagedSkillSummaries | filter:skillFilterText" ng-value="skill.id" aria-label="<[skill.description]>">
+                <[skill.description]>
```
You changed the logic here a bit, could you please explain the changes?

nishantwrp replied:
Ok So before my changes `categorizedSkills` was a dict with keys as topic names and there was one extra key `untriaged` which contained the untriaged skills. `hasUntriagedSkills` checks if the `topicName === 'untriaged'`. So what this did was iterated through the complete `categorizedSkills` and displayed the one with `untriaged` as key.

I refactored the `categorizedSkills` to not contain that extra key. Instead use a separate variable `untriagedSkillSummaries` for that purpose.

vojtechjelinek commented on core/templates/domain/opportunity/ExplorationOpportunitySummaryObjectFactory.ts:
```diff
@@ -20,17 +20,17 @@
 import { Injectable } from '@angular/core';
 
-export interface ITranslationCountsDict {
+export interface TranslationCountsDict {
```
Maybe `TranslationCountsBackendDict`?

nishantwrp replied:
Actually this is a type which is also used in the domain object. So, adding backend would suggest that it needs to be converted to a domain object which is not the case here.

vojtechjelinek commented on core/templates/domain/statistics/state-top-answers-stats-object.factory.ts:
```diff
@@ -0,0 +1,73 @@
+import { AnswerStatsObjectFactory, AnswerStats, AnswerStatsBackendDict } from
+  'domain/exploration/AnswerStatsObjectFactory';
+
+interface StateInteractionIds {
```
Maybe `StateInteractionIdsBackendDict`.

nishantwrp replied:
Actually this is a type which is also used in the domain object. So, adding backend would suggest that it needs to be converted to a domain object which is not the case here.

vojtechjelinek replied:
Okay, makes sense

vojtechjelinek commented on core/templates/domain/statistics/state-top-answers-stats-object.factory.ts:
```diff
@@ -0,0 +1,73 @@
+  [state: string]: AnswerStats[];
+}
+
+interface StateTopAnswerBackendDicts {
```
Maybe `StateTopAnswersBackendDict`.

nishantwrp replied:
done

vojtechjelinek commented on core/templates/domain/statistics/state-top-answers-stats-object.factory.ts:
```diff
@@ -0,0 +1,73 @@
+
+export interface StateTopAnswersStatsBackendDict {
+  'interaction_ids': StateInteractionIds;
+  'answers': StateTopAnswerBackendDicts;
```
```suggestion
  'answers': StateTopAnswerBackendDicts;
  'interaction_ids': StateInteractionIds;
```

nishantwrp replied:
done

vojtechjelinek commented on core/templates/domain/statistics/state-top-answers-stats-object.factory.ts:
```diff
@@ -0,0 +1,73 @@
+      backendDict: StateTopAnswersStatsBackendDict): StateTopAnswersStats {
+    let stateTopAnswers: StateTopAnswers = {};
+
+    for (var stateName in backendDict.answers) {
```
```suggestion
    for (let stateName in backendDict.answers) {
```

nishantwrp replied:
done

vojtechjelinek commented on core/templates/domain/subtopic_viewer/subtopic-viewer-backend-api.service.ts:
```diff
@@ -20,8 +20,11 @@ import { downgradeInjectable } from '@angular/upgrade/static';
+import {
+  SubtopicDataBackendDict,
+  ReadOnlySubtopicPageData,
+  ReadOnlySubtopicPageObjectFactory
```
```suggestion
  ReadOnlySubtopicPageData,
  ReadOnlySubtopicPageObjectFactory,
  SubtopicDataBackendDict
```

nishantwrp replied:
done

vojtechjelinek commented on core/templates/domain/topic/TopicObjectFactory.ts:
```diff
@@ -200,7 +200,8 @@ export class Topic {
     let topicSkillIds = cloneDeep(
-      this._uncategorizedSkillSummaries.map((skillSummary: SkillSummary) =>{
+      this._uncategorizedSkillSummaries.map((
+          skillSummary: ShortSkillSummary) =>{
```
```suggestion
          skillSummary: ShortSkillSummary) => {
```

nishantwrp replied:
done

vojtechjelinek commented on core/templates/domain/topic/TopicObjectFactory.ts:
```diff
@@ -440,7 +443,7 @@ export class Topic {
   }
 
-  getUncategorizedSkillSummaries(): Array<SkillSummary> {
+  getUncategorizedSkillSummaries(): Array<ShortSkillSummary> {
```
Why not 
```suggestion
  getUncategorizedSkillSummaries(): ShortSkillSummary[] {
```

nishantwrp replied:
done

vojtechjelinek commented on core/templates/domain/skill/ShortSkillSummaryObjectFactory.ts:
```diff
@@ -20,32 +20,39 @@
-export class SkillSummary {
-  _id: string;
-  _description: string;
+export interface ShortSkillSummaryBackendDict {
```
Why `Short`?

nishantwrp replied:
https://github.com/oppia/oppia/pull/9917#issuecomment-659453697

vojtechjelinek commented on core/templates/domain/topic_viewer/topic-viewer-backend-api.service.ts:
```diff
@@ -20,10 +20,13 @@ import { downgradeInjectable } from '@angular/upgrade/static';
+import {
+  ReadOnlyTopicObjectFactory,
+  ReadOnlyTopic,
+  ReadOnlyTopicBackendDict
```
```suggestion
  ReadOnlyTopic,
  ReadOnlyTopicBackendDict,
  ReadOnlyTopicObjectFactory
```

nishantwrp replied:
done

vojtechjelinek commented on core/templates/domain/topics_and_skills_dashboard/topics-and-skills-dashboard-backend-api.service.ts:
```diff
@@ -21,106 +21,179 @@ import { downgradeInjectable } from '@angular/upgrade/static';
+import {
+  AssignedSkillObjectFactory,
+  AssignedSkill,
+  AssignedSkillBackendDict
```
```suggestion
  AssignedSkill,
  AssignedSkillBackendDict,
  AssignedSkillObjectFactory
```

nishantwrp replied:
done

vojtechjelinek commented on core/templates/domain/topics_and_skills_dashboard/topics-and-skills-dashboard-backend-api.service.ts:
```diff
@@ -21,106 +21,179 @@ import { downgradeInjectable } from '@angular/upgrade/static';
+import {
+  AugmentedSkillSummaryObjectFactory,
+  AugmentedSkillSummaryBackendDict,
+  AugmentedSkillSummary
```
```suggestion
  AugmentedSkillSummary,
  AugmentedSkillSummaryBackendDict,
  AugmentedSkillSummaryObjectFactory
```

nishantwrp replied:
done

vojtechjelinek commented on core/templates/domain/topics_and_skills_dashboard/topics-and-skills-dashboard-backend-api.service.ts:
```diff
@@ -21,106 +21,179 @@ import { downgradeInjectable } from '@angular/upgrade/static';
+import {
+  ShortSkillSummaryBackendDict,
+  ShortSkillSummary,
+  ShortSkillSummaryObjectFactory
```
```suggestion
  ShortSkillSummary,
  ShortSkillSummaryBackendDict,
  ShortSkillSummaryObjectFactory
```

nishantwrp replied:
done

vojtechjelinek commented on core/templates/domain/topics_and_skills_dashboard/topics-and-skills-dashboard-backend-api.service.ts:
```diff
@@ -21,106 +21,179 @@ import { downgradeInjectable } from '@angular/upgrade/static';
+  ShortSkillSummary,
+  ShortSkillSummaryObjectFactory
+} from 'domain/skill/ShortSkillSummaryObjectFactory';
+import { SkillSummaryObjectFactory, SkillSummary, SkillSummaryBackendDict } from
```
```suggestion
import { SkillSummary, SkillSummaryBackendDict, SkillSummaryObjectFactory } from
```

nishantwrp replied:
done

vojtechjelinek commented on core/templates/domain/topics_and_skills_dashboard/topics-and-skills-dashboard-backend-api.service.ts:
```diff
@@ -21,106 +21,179 @@ import { downgradeInjectable } from '@angular/upgrade/static';
   TopicsAndSkillsDashboardFilter
 // eslint-disable-next-line max-len
 } from 'domain/topics_and_skills_dashboard/TopicsAndSkillsDashboardFilterObjectFactory';
+import { TopicSummaryObjectFactory, TopicSummary, TopicSummaryBackendDict } from
```
```suggestion
import { TopicSummary, TopicSummaryBackendDict, TopicSummaryObjectFactory } from
```

nishantwrp replied:
done

vojtechjelinek commented on core/templates/pages/community-dashboard-page/services/contribution-opportunities-backend-api.service.ts:
```diff
@@ -21,26 +21,61 @@ import { downgradeInjectable } from '@angular/upgrade/static';
+  FeaturedTranslationLanguage,
   FeaturedTranslationLanguageObjectFactory,
-  IFeaturedTranslationLanguageBackendDict,
+  FeaturedTranslationLanguageBackendDict,
```
```
  FeaturedTranslationLanguage,
  FeaturedTranslationLanguageBackendDict,
  FeaturedTranslationLanguageObjectFactory
```

nishantwrp replied:
done

vojtechjelinek commented on core/templates/pages/profile-page/profile-page-backend-api.service.ts:
```diff
@@ -26,6 +26,8 @@ import { ProfilePageDomainConstants } from
 import { UrlInterpolationService } from
   'domain/utilities/url-interpolation.service';
 import { UrlService } from 'services/contextual/url.service.ts';
+import { UserProfileObjectFactory, UserProfile, UserProfileBackendDict } from
```
```suggestion
import { UserProfile, UserProfileBackendDict, UserProfileObjectFactory } from
```

nishantwrp replied:
done

vojtechjelinek commented on core/templates/pages/profile-page/profile-page.component.ts:
```diff
@@ -57,35 +59,40 @@ angular.module('oppia').component('profilePage', {
           }];
 
-          ctrl.userEditedExplorations = data.edited_exp_summary_dicts.sort(
+          ctrl.userEditedExplorations = data.editedExpSummaries.sort(
```
Thanks for fixing this!

vojtechjelinek commented on core/templates/pages/skill-editor-page/services/question-creation.service.ts:
```diff
@@ -40,7 +40,7 @@ require('domain/skill/MisconceptionObjectFactory.ts');
 require('domain/skill/skill-creation-backend-api.service.ts');
 require('domain/skill/SkillDifficultyObjectFactory.ts');
-require('domain/skill/SkillSummaryObjectFactory.ts');
+require('domain/skill/ShortSkillSummaryObjectFactory.ts');
```
This should be move before `require('domain/skill/skill-backend-api.service.ts');`.

nishantwrp replied:
done

vojtechjelinek commented on core/templates/pages/topic-editor-page/subtopic-editor/subtopic-editor-tab.component.spec.ts:
```diff
@@ -56,7 +56,7 @@ describe('Subtopic editor tab', function() {
     SubtopicObjectFactory = $injector.get('SubtopicObjectFactory');
     SubtopicPageObjectFactory = $injector.get('SubtopicPageObjectFactory');
-    SkillSummaryObjectFactory = $injector.get('SkillSummaryObjectFactory');
+    SkillSummaryObjectFactory = $injector.get('ShortSkillSummaryObjectFactory');
```
```suggestion
    ShortSkillSummaryObjectFactory = $injector.get('ShortSkillSummaryObjectFactory');
```

nishantwrp replied:
done

vojtechjelinek commented on core/templates/services/state-top-answers-stats-backend-api.service.ts:
```diff
@@ -21,6 +21,11 @@ import { HttpClient } from '@angular/common/http';
+import {
+  StateTopAnswersStatsBackendDict,
+  StateTopAnswersStats,
+  StateTopAnswersStatsObjectFactory
```
```suggestion
  StateTopAnswersStats,
  StateTopAnswersStatsBackendDict,
  StateTopAnswersStatsObjectFactory
```

nishantwrp replied:
done

vojtechjelinek commented on core/templates/services/state-top-answers-stats.service.ts:
```diff
@@ -100,17 +100,16 @@ angular.module('oppia').factory('StateTopAnswersStatsService', [
         }
         workingStateTopAnswersStats = {};
-        for (var stateName in stateTopAnswersStatsBackendDict.answers) {
+        for (var stateName in stateTopAnswersStats.answers) {
```
```suggestion
        for (let stateName in stateTopAnswersStats.answers) {
```

nishantwrp replied:
done

vojtechjelinek reviewed (changes requested):
Thanks! Few minor comments and you bigger one: Why did you change `SkillSummaryObjectFactory` to `ShortSkillSummaryObjectFactory`?

nishantwrp committed:
reviews

nishantwrp commented:
> Why did you change SkillSummaryObjectFactory to ShortSkillSummaryObjectFactory

@vojtechjelinek I changed `SkillSummary` to `ShortSkillSummary` because the actual `SkillSummary` (referring to the one in backend) had more info than just `id` and `description`. So, I created a new `SkillSummary` which matched with the backend SkillSummaryDict. And renamed the current one.

UPDATE - To avoid any confusions the new `SkillSummary` is also used in the `TopicsAndSkillsDashboard`

nishantwrp commented:
@vojtechjelinek ptal!

nishantwrp committed:
Merge remote-tracking branch 'upstream/develop' into refactor-backend-api-services-3

nishantwrp commented:
@vojtechjelinek ptal!

vojtechjelinek reviewed (approved):
LGTM!

sagangwee reviewed (approved):
@nishantwrp LGTM for codeowners. Thanks!

nishantwrp commented:
@seanlip @kevintab95 @nithusha21 @aks681 @bansalnitish @srijanreddy98 @Showtim3 ptal!

seanlip commented on core/templates/domain/statistics/state-top-answers-stats-object.factory.ts:
```diff
@@ -0,0 +1,73 @@
+  'domain/exploration/AnswerStatsObjectFactory';
+
+interface StateInteractionIds {
+  [state: string]: string;
```
Should be stateName not state, right? Ditto elsewhere.

Be careful to distinguish between these two concepts.

nishantwrp replied:
done

seanlip commented on core/templates/pages/profile-page/profile-page.component.spec.ts:
```diff
@@ -166,19 +190,156 @@ describe('Profile page', function() {
+        title: 'Test Title'
+      });
+      profileData.edited_exp_summary_dicts.push({
+        last_updated_msec: 1591296737470.528,
```
I wonder if there's any way you can think of to reduce the duplication a bit, here and elsewhere in this file. Otherwise, the important information about the differences between the different dicts gets lost in the overall jumble of information. Perhaps you could use a for loop.

nishantwrp replied:
does it look good now?

seanlip replied:
Yes, I think so, thanks.

seanlip commented on core/templates/pages/profile-page/profile-page.component.ts:
```diff
@@ -96,10 +103,10 @@ angular.module('oppia').component('profilePage', {
           );
 
-          ctrl.userNotLoggedIn = !data.username;
+          ctrl.userNotLoggedIn = !data.currentUser;
```
This seems like a functionality change. Where is currentUser defined?

If it is an error, why was it not caught?

seanlip replied:
(Note: this is probably covered by the comment below.)

nishantwrp replied:
Yes it's defined in `user-profile-object.factory.ts`.

seanlip commented on core/templates/domain/user/user-profile-object.factory.ts:
```diff
@@ -0,0 +1,92 @@
+
+  createFromBackendDict(backendDict: UserProfileBackendDict) {
+    return new UserProfile(
+      backendDict.username,
```
Woah. Why do we have "username" and "profile username"? That's just confusing.

Are both needed? What is the difference between the two fields?

Also you should not be renaming this to "currentUser". Please be really careful about the names of things. A "user" and a "username" are different. I've seen this pattern in a couple places now so I'd like to request that you audit all the fields and feel free to ask questions if you are not sure whether a renaming is valid.

nishantwrp replied:
profile usename -> the username of profile being visited.
username -> This is the username of the user currently logged in. This is sent along with almost every endpoint.

> Are both needed

This is used in that page just to determine if any user is logged in.

Should I change it to `currentUsername` instead?

seanlip replied:
username and usernameOfViewedProfile should work, I think. Best to keep "username" consistent everywhere.

Also, you have not replied to my last paragraph. Please make sure to respond to **all** parts in a comment. I would like a confirmation that you have reviewed every field -- probably what makes the most sense is for you to please list all the name changes explicitly here, with links to the code for each of those.

nishantwrp replied:
> username and usernameOfViewedProfile should work, I think. Best to keep "username" consistent everywhere.

done.

Ah, sorry! I think I missed the last para. I had some doubts with it.

> I've seen this pattern in a couple places now so I'd like to request that you audit all the fields and feel free to ask questions if you are not sure whether a renaming is valid.

I didn't actually understand which pattern?

seanlip replied:
The pattern of renaming fields to different names (i.e. anywhere where there's any sort of difference between the key name in the backend dict, the attribute name of the frontend object, or the variable used to refer to a value in the dict/object).

Does that clarify?

nishantwrp replied:
@seanlip i had some questions
1. does that mean that in some cases like this one to change the name of field in backend_dict too? (changing profile_username to username_of_viewed_profile)
2. In some cases the backend_dict says skill_id and in the domain it is changed to id. Should these kind of cases be also changed?

seanlip replied:
Renaming is fine in some cases as long as it is not confusing. Hence the request for a list of all instances (with links to the relevant lines of code) so that I can give you specific advice.

nishantwrp replied:
Here are instances I could find

https://github.com/oppia/oppia/blob/f3ec0b4e15059ab8e6b452abca8bb9e59b6292bf/core/templates/domain/topic_viewer/read-only-topic-object.factory.ts#L38-L61

Here properties like `additional_story_dicts` is renamed to `additionalStorySummaries`. I personally think, this is okay!


This file -> https://github.com/oppia/oppia/blob/develop/core/templates/domain/user/profile-summary-object.factory.ts
This is a common object factory for two backend dicts because it doesn't make sense to make a separate objects for these dicts.

Here -> https://github.com/oppia/oppia/blob/develop/core/templates/domain/exploration/RuleObjectFactory.ts
`rule_type` is renamed to just `type`. 

Here -> https://github.com/oppia/oppia/blob/develop/core/templates/domain/question/QuestionSummaryObjectFactory.ts
I've corrected this one!

Here -> https://github.com/oppia/oppia/blob/develop/core/templates/domain/skill/SkillSummaryObjectFactory.ts
I think thats okay!

Here -> https://github.com/oppia/oppia/blob/develop/core/templates/domain/skill/SkillRightsObjectFactory.ts
Corrected this one!

Here -> https://github.com/oppia/oppia/blob/develop/core/templates/domain/feedback_thread/FeedbackThreadSummaryObjectFactory.ts
Corrected this one!

nishantwrp replied:
@seanlip ptal!

seanlip reviewed (changes requested):
Thanks @nishantwrp, took a pass at codeowner files.

nishantwrp committed:
reviews

seanlip commented on core/templates/pages/profile-page/profile-page.component.spec.ts:
```diff
@@ -214,129 +214,54 @@ describe('Profile page', function() {
-        category: 'Algebra',
-        title: 'Test Title'
-      });
+      for (let i = 0; i<11; i++) {
```
Add spaces around <

nishantwrp replied:
done

nishantwrp committed:
reviews

nishantwrp committed:
reviews

bansalnitish reviewed (approved):
LGTM for codeowner files.

srijanreddy98 reviewed (approved):
LGTM for code owner files!

kevintab95 reviewed (approved):
LGTM for codeowner files. Thanks @nishantwrp!

aks681 reviewed (approved):
Lgtm as codeowner

seanlip commented:
Looks good -- just the one comment left: https://github.com/oppia/oppia/pull/9917#discussion_r456836049

Also @nithusha21 PTAL as codeowner.

nithusha21 reviewed (approved):
LGTM as a codeowner!

nishantwrp committed:
reviews

seanlip commented on core/templates/domain/question/QuestionSummaryObjectFactory.ts:
```diff
@@ -26,17 +26,17 @@ export interface IQuestionSummaryBackendDict {
 
 export class QuestionSummary {
-  _questionId: string;
+  _id: string;
```
Leave this as it was (_questionId) and perhaps update the backend dict accordingly. Better to be clearer about what entity has the id (the question, not the question summary).

nishantwrp replied:
>Better to be clearer about what entity has the id (the question, not the question summary).

Got it! done!

seanlip reviewed (changes requested):
Thanks @nishantwrp, everything looks good to me except for one comment. I agree with your other design decisions.

nishantwrp committed:
reviews

nishantwrp committed:
Merge remote-tracking branch 'upstream/develop' into refactor-backend-api-services-3

nishantwrp committed:
Fix issues with merge

seanlip reviewed (approved):
Thanks! LGTM.

seanlip commented:
@nishantwrp could you please update https://github.com/oppia/oppia/issues/8015 accordingly with what has been done? Thanks!

nishantwrp commented:
@seanlip done!

seanlip commented:
@nishantwrp did you move them to "Fixed part of the issue"? If so please add your username to the ones you did and ideally the PR number so that it's easier to track.

seanlip commented:
Note: for clarity, I have also renamed that section to "Completed files (for reference)".
